esphome/components/modbus/modbus.cpp

```diff
@@ -11,46 +13,77 @@ void Modbus::setup() {
   if (this->flow_control_pin_ != nullptr) {
     this->flow_control_pin_->setup();
   }
+  // 3.5 characters * 11 bits per character * 1000ms/sec / (bits/sec) (Standard modbus frame delay)
+  this->frame_delay_ms_ = (3.5 * 11 * 1000 / this->parent_->get_baud_rate()) + 1;
```

Comment:
Round up instead of adding 1 ms.
```suggestion
  this->frame_delay_ms_ = ((35 * 11 * 1000 / this->parent_->get_baud_rate()) + 5 ) / 10;
```